### PR TITLE
Change the name of the balancer proto service from Msg -> BalancerMsg

### PR DESCRIPTION
**Pull request by ValarDragon**
This previously was causing problems for typescript proto generation,
because we have Balancer's proto Msg service and gamm's proto Msg service
existing at the same proto import path.

They existed at the same import path, because due to interface serialization
depending on the amino route (which depended on the proto import path),
changing the import path requires a state migration thats a bit cumbersome to
do.

Closes: #1185 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: proto/osmosis/txfees/v1beta1/query.proto [hunk 1]
```diff
@@ -19,7 +19,8 @@ service Query {
 
   rpc DenomSpotPrice(QueryDenomSpotPriceRequest)
       returns (QueryDenomSpotPriceResponse) {
-    option (google.api.http).get = "/osmosis/txfees/v1beta1/spot_price_by_denom";
+    option (google.api.http).get =
+        "/osmosis/txfees/v1beta1/spot_price_by_denom";
```
Comment (ValarDragon) [inline proto/osmosis/txfees/v1beta1/query.proto:23]: done by proto formatting automatically

File: x/gamm/pool-models/balancer/tx.pb.go [hunk 2]
```diff
@@ -188,72 +188,72 @@ var _ grpc.ClientConn
...
+		return srv.(BalancerMsgServer).CreateBalancerPool(ctx, req.(*MsgCreateBalancerPool))
 	}
 	return interceptor(ctx, in, info, handler)
 }
 
-var _Msg_serviceDesc = grpc.ServiceDesc{
-	ServiceName: "osmosis.gamm.v1beta1.Msg",
-	HandlerType: (*MsgServer)(nil),
+var _BalancerMsg_serviceDesc = grpc.ServiceDesc{
+	ServiceName: "osmosis.gamm.v1beta1.BalancerMsg",
```
Comment (ValarDragon) [inline x/gamm/pool-models/balancer/tx.pb.go:251]: Is this service name at all important? 

an alternative approach to fixing this problem entirely, is to migrate the balancer pools and not deal with this duplicate import situation

Comment (mattverse) [reply x/gamm/pool-models/balancer/tx.pb.go:251]: why is still it creating BalancerMsg under v1beta1?  I was assuming it should be created as something like `osmosis.gamm.pool-models.BalacerMsg`

Wdym by migrating balancer pools?

File: proto/osmosis/gamm/pool-models/balancer/tx.proto [hunk 3]
```diff
@@ -6,7 +6,7 @@ import "osmosis/gamm/pool-models/balancer/balancerPool.proto";
 
 option go_package = "github.com/osmosis-labs/osmosis/v7/x/gamm/pool-models/balancer";
 
-service Msg {
+service BalancerMsg {
```
Comment (alexanderbez) [inline proto/osmosis/gamm/pool-models/balancer/tx.proto:9]: Can you explain how/where this service is used? Does this actually process on-chain messages?

**Review by alexanderbez — APPROVED** [review]

File: x/gamm/pool-models/balancer/tx.pb.go (hunk 2 above)
Comment (ValarDragon) [reply x/gamm/pool-models/balancer/tx.pb.go:251]: oh thats a good point, we could make txs under the correct package, but just leave the pool proto definition under the legacy wrong one

File: proto/osmosis/gamm/pool-models/balancer/tx.proto (hunk 3 above)
Comment (ValarDragon) [reply proto/osmosis/gamm/pool-models/balancer/tx.proto:9]: It should process on-chain messages at least, I have not confirmed on a testnet. The service name as far as I can tell is just for generated protobuf structs, we have methods to register the proto service, linking the go struct to the proto message

File: x/gamm/pool-models/balancer/tx.pb.go (hunk 2 above)
Comment (ValarDragon) [reply x/gamm/pool-models/balancer/tx.pb.go:251]: I think your proposal is a better approach

File: proto/osmosis/gamm/pool-models/balancer/tx.proto (hunk 3 above)
Comment (alexanderbez) [reply proto/osmosis/gamm/pool-models/balancer/tx.proto:9]: Sorry if this is obvious, but why are they not all under a single `service?

Comment (mattverse) [reply proto/osmosis/gamm/pool-models/balancer/tx.proto:9]: https://github.com/bufbuild/buf/issues/99#issuecomment-655841579

**Comment by ValarDragon** [comment]
Going to close this as stale for now. Lets update with the fix matt pointed out

**Pull request closed without merging**